pkg/controller/drain.go

```diff
@@ -313,13 +314,12 @@ func (o *DrainOptions) deleteOrEvictPods(pods []api.Pod) error {
 }
 
 func (o *DrainOptions) evictPods(pods []api.Pod, policyGroupVersion string, getPodFn func(namespace, name string) (*api.Pod, error)) error {
-	doneCh := make(chan bool, len(pods))
-	errCh := make(chan error, 1)
+	returnCh := make(chan error, 1)
```

Comment:
Should the buffered channel be initialized to len(pods) , as there could be more elements in it?

Reply:
This logic is taken from:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/drain/drain.go#L354

I guess we would end up serialising the "exiting of Go routines" also, instead of serialising just the "incrementing `doneCount`".

Reply:
I see. Not sure, at the lowest level how golang would behave, but I understand if by any chance more than one elements are piped into the channel, it would panic. 
Let's keep it this way then.